src/effects/presets/effectpreset.cpp

```diff
@@ -99,5 +99,22 @@ const QDomElement EffectPreset::toXml(QDomDocument* doc) const {
     return effectElement;
 }
 
-EffectPreset::~EffectPreset() {
+void EffectPreset::updateParametersFrom(const EffectPreset& other) {
+    DEBUG_ASSERT(backendType() == other.backendType());
+
+    // technically algorithmically inefficient solution O(n²). May be
+    // optimizable by sorting first, gains depend on parameter count
+    for (const auto& parameterToCopy : other.m_effectParameterPresets) {
+        auto currentParameterIt =
+                std::find_if(m_effectParameterPresets.begin(),
+                        m_effectParameterPresets.end(),
+                        [&](const auto& ourPreset) {
+                            return ourPreset.id() == parameterToCopy.id();
+                        });
+        if (currentParameterIt == m_effectParameterPresets.cend()) {
+            continue;
+        }
+        // overwrite our parameter by taking a copy of the same parameter from `other`
+        *currentParameterIt = parameterToCopy;
```

Comment:
Actually I just realized, does this create a dangling reference? How do I force a copy here so `m_effectParameterPresets` takes ownership?
Does this suffice?
```suggestion
        // overwrite our parameter by taking a copy of the same parameter from `other`
        *currentParameterIt = EffectParameterPreset(parameterToCopy);
```

Reply:
I think the proposed version is redundant because the copy constructor is already used.

Reply:
Thinking more about it, I'm pretty sure its the copy-assignment operator, so we should be good then.